### DISCUSSION TITLE
How to get coredump from inside docker container

### DISCUSSION THREAD

**Discussion by julianoes** [community/community#25136] [Actions]
<p>I would like to collect coredumps from a C++ binary running inside a docker container in GitHub workflows in order to first print a backtrace immediately and second store it as an artifact (together with the binary) to debug later.</p>
<p>I have set:</p>
<pre><code>ulimit -c unlimited
</code></pre>
<p>I’ve tried to set the path where core files are written:</p>
<pre><code>/cores/core.%e.%p | /proc/sys/kernel/core_pattern
</code></pre>
<p> but it fails because this is read only.</p>
<p>Similarly, this doesn’t seem to have any effect:</p>
<pre><code>sysctl -w kernel.core_pattern=core
</code></pre>
<p>In the default core dump location </p>
<pre><code>/usr/share/apport
</code></pre>
<p>is nothing to be found after a crash.</p>
<p>From reading about this my guess is that the docker container would have to run with</p>
<pre><code>--privileged
</code></pre>
<p>in order to allow changing the path that but supposedly that’s not possible.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/julianoes">@julianoes</a> ,</p>
<p>Maybe you can try using the syntax  <strong>jobs.&lt;job_id&gt;.container.options</strong> to configure the optiones.</p>
<p>For example:</p>
<pre><code>jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    container:
      image: &lt;image&gt;
      options: --privileged --ulimit core=-1 --security-opt seccomp=unconfined
    steps:
      - name: Core Dump
        run: |
          ulimit -c unlimited
          echo '/tmp/core.%h.%e.%t' &gt; /proc/sys/kernel/core_pattern
</code></pre>

**Comment by julianoes** [comment]
<p>Awesome, I tested this and it works. Thank you!</p>
